src/flowcean/transforms/match_sampling_rate.py

```diff
@@ -13,99 +13,166 @@ class MatchSamplingRate(Transform):
-    present in the DataFrame. To remove it use the `select` transform.
-    """
+    """  # noqa: E501
 
     def __init__(
         self,
-        reference_timestamps: str,
-        feature_columns_with_timestamps: dict[str, str],
+        reference_feature: str,
+        feature_columns: dict[str, str],
     ) -> None:
         """Initialize the MatchSamplingRate transform.
 
         Args:
-            reference_timestamps: Timestamps of the reference feature.
-            feature_columns_with_timestamps: Names of the features that are
-                getting interpolated with their respective original timestamp
-                feature names.
+            reference_feature: Reference timeseries feature.
+            feature_columns: key-value pairs of feature columns and
+                the interpolation method to use. The interpolation
+                method can be one of the following:
+                - 'linear'
+                - 'slerp' (to be implemented)
         """
-        self.reference_timestamps = reference_timestamps
-        self.feature_columns_with_timestamps = feature_columns_with_timestamps
+        self.reference_feature = reference_feature
+        self.feature_columns = feature_columns
 
     def transform(self, data: pl.DataFrame) -> pl.DataFrame:
+        """Transform the input DataFrame.
+
+        Args:
+            data: Input DataFrame.
+
+        Returns:
+            Transformed DataFrame.
+
+        """
+        # preserve all constant columns that are not timeseries data
+        non_timeseries_features = [
+            col
+            for col in data.columns
+            if col not in [*self.feature_columns, self.reference_feature]
+        ]
         logger.debug("Matching sampling rate of time series.")
 
-        for i in range(len(data)):
-            reference_timestamps = data[self.reference_timestamps][i]
-            for (
-                feature,
-                timestamp,
-            ) in self.feature_columns_with_timestamps.items():
-                timestamps = data[timestamp][i].to_numpy()
-                feature_data = data[feature][i].to_numpy()
-                resampled_timeseries = interp(
-                    reference_timestamps,
-                    timestamps,
-                    feature_data,
-                ).tolist()
-                data = (
-                    data.lazy()
-                    .with_row_index()
-                    .with_columns(
-                        pl.when(pl.col("index") == i)
-                        .then(resampled_timeseries)
-                        .otherwise(pl.col(feature))
-                        .alias(feature)
-                    )
-                    .drop("index")
-                    .collect()
+        if self.reference_feature not in data.columns:
+            msg = f"Reference {self.reference_feature} not found in the DataFrame."  # noqa: E501
+            raise ValueError(msg)
```

Comment:
You can reformat the string or even better create a new exception class that properly describes your error. Something similar to the `NotLoadedError`

Reply:
done